components/requeue/requeue_test.go

```diff
@@ -0,0 +1,97 @@
+package requeue
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/ThreeDotsLabs/watermill/message/router/middleware"
+	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRequeue(t *testing.T) {
+	logger := watermill.NewStdLogger(false, false)
+
+	pubSub := gochannel.NewGoChannel(gochannel.Config{}, logger)
+
+	requeue, err := NewRequeue(Config{
+		Subscriber: pubSub,
+		Topic:      "requeue",
+		Publisher:  pubSub,
+		Delay:      time.Millisecond * 200,
+	}, logger)
+	require.NoError(t, err)
+
+	go func() {
+		err := requeue.Run(context.Background())
+		require.NoError(t, err)
+	}()
+
+	router, err := message.NewRouter(message.RouterConfig{}, logger)
+	require.NoError(t, err)
+
+	pq, err := middleware.PoisonQueue(pubSub, "requeue")
```

Comment:
It would be nice to do something like: `pq, err := middleware.PoisonQueue(pubSub, requeue.Topic())`

Reply:
Might not be possible anymore since I change the config to include a topic generator. :)